File: python/backend/src/app/models/notes.py
"""Pydantic models for notes API."""

from __future__ import annotations

from pydantic import BaseModel


class NoteCreateRequest(BaseModel):
    vault_path: str
    folder: str
    filename: str
    content: str


class NoteReadRequest(BaseModel):
    vault_path: str
    path: str


class NoteUpdateRequest(BaseModel):
    vault_path: str
    path: str
    content: str


class NoteDeleteRequest(BaseModel):
    vault_path: str
    path: str


class NoteOperationResult(BaseModel):
    success: bool
    path: str | None = None
    content: str | None = None
    error: str | None = None

File: packages/shared-types/src/notes.ts
export interface INoteCreateRequest {
  vaultPath: string;
  folder: string;
  filename: string;
  content: string;
}

export interface INoteReadRequest {
  vaultPath: string;
  path: string;
}

export interface INoteUpdateRequest {
  vaultPath: string;
  path: string;
  content: string;
}

export interface INoteDeleteRequest {
  vaultPath: string;
  path: string;
}

export interface INoteOperationResult {
  success: boolean;
  path?: string;
  content?: string;
  error?: string;
}

export interface INoteMetadata {
  path: string;
  folder: string;
  filename: string;
  size: number;
  modified: number;
  tags: string[];
  links: string[];
}
